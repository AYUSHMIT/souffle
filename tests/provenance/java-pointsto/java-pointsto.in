setdepth 6
explain VarPointsTo("java.lang.Object.<init>()V|@this", "A.method1()V|new X|0")
