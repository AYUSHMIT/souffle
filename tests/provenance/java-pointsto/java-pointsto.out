                                                                                                                                                                                                                                                                                                                                                         SpecialMethodInvocationBase("A.method1()V|X.<init>()V|0", "A.method1()V|%1")  SpecialMethodInvocationSignature("A.method1()V|X.<init>()V|0", "<X.<init>()V>")                                                                                                                                      
                                                                                                                                                                                                                                                                                                                                                         -------------------------------------------------------------------------(R2) ----------------------------------------------------------------------------(R1)                                                                                                                                     
                                                                                                                                                                                                                                                                                                                                                                InvocationBase("A.method1()V|X.<init>()V|0", "A.method1()V|%1")                   Invocation("A.method1()V|X.<init>()V|0", "<X.<init>()V>")             ThisVar("<X.<init>()V>", "X.<init>()V|@this")  AssignHeapAllocation("A.method1()V|new X|0", "A.method1()V|%1", "<A.method1()V>")    
                                                                                                                                                                                                                                                                                                                                                         ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------(R4) ------------------------------------------------------------------------------(R1)   
SpecialMethodInvocationBase("X.<init>()V|java.lang.Object.<init>()V|0", "X.<init>()V|0")  SpecialMethodInvocationSignature("X.<init>()V|java.lang.Object.<init>()V|0", "<java.lang.Object.<init>()V>")                                                                               AssignLocal("X.<init>()V|@this", "X.<init>()V|0", "<X.<init>()V>")                                                                                 Assign("A.method1()V|%1", "X.<init>()V|@this")                                                                                               VarPointsTo("A.method1()V|%1", "A.method1()V|new X|0")                 
-------------------------------------------------------------------------------------(R2) ---------------------------------------------------------------------------------------------------------(R1)                                                                              ---------------------------------------------------------------(R1) ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------(R2)  
       InvocationBase("X.<init>()V|java.lang.Object.<init>()V|0", "X.<init>()V|0")                   Invocation("X.<init>()V|java.lang.Object.<init>()V|0", "<java.lang.Object.<init>()V>")             ThisVar("<java.lang.Object.<init>()V>", "java.lang.Object.<init>()V|@this")             Assign("X.<init>()V|@this", "X.<init>()V|0")                                                                                                                                 VarPointsTo("X.<init>()V|@this", "A.method1()V|new X|0")                                                                                                                       
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------(R4) ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------(R2) 
                                                                                                            Assign("X.<init>()V|0", "java.lang.Object.<init>()V|@this")                                                                                                                                                                                                                                                                       VarPointsTo("X.<init>()V|0", "A.method1()V|new X|0")                                                                                                                                                          
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------(R2)
                                                                                                                                                                                                                                                                                          VarPointsTo("java.lang.Object.<init>()V|@this", "A.method1()V|new X|0")                                                                                                                                                                                                                                                                                           
